docs/docsite/rst/dev_guide/developing_modules_general_windows.rst

```diff
@@ -0,0 +1,261 @@
+- When trying an exception use ``Fail-Json -obj $result -message "exception message here"`` instead
+- Most new modules require check mode and integration tests before they are merged into the main Ansible codebase
+- Avoid using try/catch statements over a bit code block, rather use then for individual calls so the error message can be more descriptive
+- Avoid using PSCustomObjects unless necessary
+- Look for common functions at ``./lib/ansible/module_utils/powershell/`` and use the code there instead of duplicating work, these can be imported by adding the line ``#Requires -Module *`` below ``#POWERSHELL_COMMON`` where * is the filename to import
+- Ensure the code is complient with Powershell v3 and above, if it isn't then document the requirements
+- Ansible uses strictmode version 2.0, be sure to test with that enabled by putting ``Set-StrictMode -Version 2.0`` at the top of your dev script
+- Favour native Powershell cmdlets instead of executable calls if possible
+- If adding a object to ``$result`` ensure any trailing slashes are removed or exited out as ``ConvertTo-Json`` will fail to convert it
+
+A very basic powershell module template can be found found below
+
+.. code-block:: powershell
+
+    #!powershell
+    # This file is part of Ansible
+
+    # GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+    # WANT_JSON
+    # POWERSHELL_COMMON
+
+    $ErrorActionPreference = 'Stop'
+
+    $params = Parse-Args -arguments $args -supports_check_mode $true
+    $check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
+    $diff_mode = Get-AnsibleParam -obj $params -name "_ansible_diff" -type "bool" -default $false
+
+    # these are your module parameters, there are various types which can be
+    # used to format your parameters. You can also set mandatory parameters
+    # with -failifempty, set defaults with -default and set choices with
+    # -validateset.
+    $string = Get-AnsibleParam -obj $params -name "string" -type "str" -failifempty $true
+    $bool = Get-AnsibleParam -obj $params -name "bool" -type "bool" -default $false
+    $int = Get-AnsibleParam -obj $params -name "int" -type "int"
+    $path = Get-AnsibleParam -obj $params -name "path" -type "path"
+    $list = Get-AnsibleParam -obj $params -name "list" -type "list"
+    $choices = Get-AnsibleParam -obj $params -name "choices" -type "str" -default "present" -validateset "absent","present"
+
+    $result = @{
+        changed = $false
+    }
+
+    if ($diff_mode) {
+        $result.diff = @{}
+    }
+
+    # code goes here
+
+    # you can add/set new result objects with
+    $result.changed = $true
+    $result.new_result = "Hi"
+
+    Exit-Json -obj $result
+
+
+When in doubt look at some of the core modules and see how things have been
+implemented there.
+
+Sometimes there are multiple ways that Windows offers to complete a task, this
+is the order to favour when writing modules
```

Comment:
Semicolon instead of comma after the word "task"; colon at the end of the sentence.